wiring/src/spark_wiring_mesh.cpp

```diff
@@ -137,37 +133,71 @@ int MeshPublish::fetchMulticastAddress(IPAddress& mcastAddr) {
     return 0;
 }
 
-int MeshPublish::initialize_udp() {
+int MeshPublish::initializeUdp() {
     std::lock_guard<RecursiveMutex> lk(mutex_);
-    if (udp) {
+    if (udp_) {
         return SYSTEM_ERROR_NONE;
     }
-    udp.reset(new UDP());
+    std::unique_ptr<UDP> udp(new UDP());
     if (!udp) {
         return SYSTEM_ERROR_NO_MEMORY;
     }
     udp->setBuffer(MAX_PACKET_LEN);
-    // Get OpenThread interface index (     interface is named "th1" on all Mesh devices)
+    // Get OpenThread interface index (interface is named "th1" on all Mesh devices)
     uint8_t idx = 0;
     if_name_to_index("th1", &idx);
-     // Create UDP socket and bind to OpenThread interface
+    // Create UDP socket and bind to OpenThread interface
     CHECK(udp->begin(PORT, idx));
-    // subscribe to multicast
 
+    // subscribe to multicast
     IPAddress mcastAddr;
     CHECK(fetchMulticastAddress(mcastAddr));
-    udp->joinMulticast(mcastAddr);
+    CHECK(udp->joinMulticast(mcastAddr));
+
+    // Start polling thread
+    exit_ = false;
+    thread_.reset(new (std::nothrow) Thread("meshpub", [](void* ptr) {
+        auto self = (MeshPublish*)ptr;
+        while (!self->exit_) {
+            self->poll();
+        }
+    }, this, OS_THREAD_PRIORITY_DEFAULT + 1));
+
+    if (!thread_) {
+        return SYSTEM_ERROR_NO_MEMORY;
+    }
+
+    udp_ = std::move(udp);
     return SYSTEM_ERROR_NONE;
 }
 
-int MeshPublish::uninitialize_udp() {
+int MeshPublish::uninitializeUdp() {
+    // NB: this function is not really used at the moment
+
+    // Use atomic exit_ as a simple synchronization primitive to ensure thread-safety
+    // instead of a mutex here.
+    if (exit_) {
```

Comment:
there's a race condition here between reading and setting the value - an atomic exchange will avoid this.

```suggestion
    if (exit_.exchange(true) {
```